File: src/main/java/com/flexpoker/web/init/WebInitializer.java
package com.flexpoker.web.init;

import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.ServletRegistration.Dynamic;

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;

import com.flexpoker.config.ForceHttpsFilter;
import com.flexpoker.config.WebConfig;
import com.flexpoker.config.WebSocketConfig;

public class WebInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {

    @Override
    public void onStartup(ServletContext container) throws ServletException {
        container.addFilter(ForceHttpsFilter.class.getSimpleName(), ForceHttpsFilter.class)
                .addMappingForUrlPatterns(null, false, "/*");
        super.onStartup(container);
    }

    @Override
    protected Class<?>[] getRootConfigClasses() {
        return null;
    }

    @Override
    protected Class<?>[] getServletConfigClasses() {
        return new Class<?>[] { WebConfig.class, WebSocketConfig.class };
    }

    @Override
    protected String[] getServletMappings() {
        return new String[] { "/" };
    }

    @Override
    protected void customizeRegistration(Dynamic registration) {
        registration.setAsyncSupported(true);
    }

}

File: src/main/kotlin/com/flexpoker/web/init/WebInitializer.kt
package com.flexpoker.web.init

import com.flexpoker.config.ForceHttpsFilter
import com.flexpoker.config.WebConfig
import com.flexpoker.config.WebSocketConfig
import jakarta.servlet.ServletContext
import jakarta.servlet.ServletRegistration
import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer

class WebInitializer : AbstractAnnotationConfigDispatcherServletInitializer() {

    override fun onStartup(container: ServletContext) {
        container.addFilter(ForceHttpsFilter::class.java.simpleName, ForceHttpsFilter::class.java)
            .addMappingForUrlPatterns(null, false, "/*")
        super.onStartup(container)
    }

    override fun getRootConfigClasses(): Array<Class<*>>? {
        return null
    }

    override fun getServletConfigClasses(): Array<Class<*>> {
        return arrayOf(WebConfig::class.java, WebSocketConfig::class.java)
    }

    override fun getServletMappings(): Array<String> {
        return arrayOf("/")
    }

    override fun customizeRegistration(registration: ServletRegistration.Dynamic) {
        registration.setAsyncSupported(true)
    }

}